### ISSUE TITLE
have separate workflows per system (or output type)

### ISSUE THREAD

**Issue by sellout** [garnix-io/issues#65]
Currently, all jobs appear under a single “Garnix CI” workflow, with output type, name, and system all in the job name:

![Screenshot 2023-07-11 at 01 41 01](https://github.com/sellout/yaya/assets/33031/cace3b8e-54cd-4820-a910-dfaa39c2187d)

This often makes it hard to see the full name.

Unfortunately, GH doesn’t allow grouping of jobs, but at least multiple workflows can be used to provide a single level of grouping. I’m not sure which makes more sense: output type or system. But it would be nice to have it look more like:

- Garnix
  - Evaluate flake.nix 
- Garnix [aarch64-darwin]
  - check format
  - check lint
  - devShell default
  - devShell ghc8107
  - package ghc961_yaya
  - package ghc961_yaya_hedgehog
- Garnix [aarch64-linux]
  - check format
  - check lint
  - devShell default
  - devShell ghc8107
  - package ghc961_yaya
  - package ghc961_yaya_hedgehog
- Garnix [x86_64-linux]
  - check format
  - check lint
  - devShell default
  - devShell ghc8107
  - package ghc961_yaya
  - package ghc961_yaya_hedgehog

or

- Garnix
  - Evaluate flake.nix 
- Garnix checks
  - format [aarch64-darwin]
  - format [aarch64-linux]
  - format [x86_64-linux]
  - lint [aarch64-darwin]
  - lint [aarch64-linux]
  - lint [x86_64-linux]
- Garnix devShells
  - default [aarch64-darwin]
  - default [aarch64-linux]
  - default [x86_64-linux]
  - ghc8107 [aarch64-darwin]
  - ghc8107 [aarch64-linux]
  - ghc8107 [x86_64-linux]
- Garnix packages
  - ghc961_yaya [aarch64-darwin]
  - ghc961_yaya [aarch64-linux]
  - ghc961_yaya [x86_64-linux]
  - ghc961_yaya-hedgehog [aarch64-darwin]
  - ghc961_yaya-hedgehog [aarch64-linux]
  - ghc961_yaya-hedgehog [x86_64-linux]

**Comment by gvolpe** [comment]
I like the first option more 🤩

**Comment by sellout** [comment]
> I like the first option more 🤩

Yeah, I agree. Especially with #57, it’s clear that builds on the same arch are more closely related.

**Comment by jkarni** [comment]
How does this work? What are workflows in the context of the github checks API?

**Comment by sellout** [comment]
Hrmm, it seems like maybe it doesn’t work 😕 Looking at the [Check Suites API](https://docs.github.com/en/rest/checks/suites?apiVersion=2022-11-28)” (sorry, workflow was the wrong term), it seems like there’s only one check suite per owner+repo+SHA. There are other interfaces that talk about filtering check suites for a SHA and check suites having names, but the creation API doesn’t have many parameters.

**Comment by sellout** [comment]
To pivot this issue a bit … now that I mostly view builds on garnix.io, I would love to see this kind of grouping there.
